---
layout: lesson
root: .
permalink: index.html  # Is the only page that don't follow the partner /:path/index.html
---

The lessons of the first two days use the
[Software Carpentry]({{ site.swc_site }}) and
[Data Carpentry]({{ site.dc_site }}) lesson template and are based on the [Data Carpentry Genomics lesson]({{ site.dc_site }}).


> ## Schedule week 1: Prokaryotic microbes (bacteria)
> 
> Days 1-2 are taught by Anita Schuerch. Days 3-4 are taught by Aldert Zomer and Bas Dutilh.
> 
> 1.  Tuesday, 3 April: Intro to data and the command line, shell scripting, Quality control, workflows
> 2.  Wednesday, 4 April: Workflows continued, Genome visualization, Metadata manipulation with R
> 3.  Thursday, 5 April: Trimming, QC, Assembly, Assembly lecture
> 4.  Friday, 6 April: Annotation, pangenome analysis, Association scoring, Wrap-up

> ## Schedule week 2: Eukaryotic microbes (fungi)
> 
> Days 5-8 are taught by Jerome Collemare, Ronnie de Jonge, and Robin Ohm.
> 
> 5.  Monday, 9 April: Assembly
> 6.  Tuesday, 10 April: RNAseq and gene prediction
> 7.  Wednesday, 11 April: Functional annotation
> 8.  Thursday, 12 April: Comparative genomics
> 9.  Friday, 13 April: Exam

> ## Data
>
> In days 1-2 we will use data from a long term evolution experiment published in 2012: [Genomic analysis of a key innovation in an experimental Escherichia coli population by Blount ZD, Barrick JE, Davidson CJ, and Lenski RE. (doi: 10.1038/nature11514)](https://www.nature.com/articles/nature11514).
>
> We are using [this](http://pad.software-carpentry.org/2018-04-03-Utrecht) collaborative document.
{: .prereq}


{% include links.md %}
