---
layout: page
title: Setup
root: .
---

Please go through all the installation steps below and make sure that
you not only installed them, but start them up to make sure they're working.
If you have any problems, don't hesitate to email the instructors to
ask for help, or arrive early on the first day to
get help.

## Windows
### Putty
You will need a terminal program to access the HPC cluster. This is already available on Mac and Linux.
- Go to the [Putty download page](http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html)
- Click on putty.exe link to download the install file
- To use it, double-click on the download file
- Follow the installation instructions, if any 

### R
 - Install R by downloading and running [this .exe file](http://cran.r-project.org/bin/windows/base/release.htm) from [CRAN](http://cran.r-project.org/index.html)
 - Also, please install [RStudio](http://www.rstudio.com/ide/download/desktop)

## MacOSX
### R
 - Install R by downloading and running [this .pkg file](http://cran.r-project.org/bin/macosx/R-latest.pkg) from [CRAN](http://cran.r-project.org/index.html)
 - Also, please install [RStudio](http://www.rstudio.com/ide/download/desktop)

## Linux
### R
 - You can download the [binary files for your distribution] from [CRAN](http://cran.r-project.org/index.html) or you can use your package manager (e.g. for Debian/Ubuntu run 
 ```
 sudo apt-get install r-base
 ```
 and for Fedora run
 ```
 sudo yum install R
 ```
 - Also, please install [RStudio](http://www.rstudio.com/ide/download/desktop)
 
 
## Linux, Windows and Mac 

### FTP Client (Filezilla)
An FTP client will help you transfer files easily between your computer and the cloud.

- Go to the [Filezilla](https://filezilla-project.org/download.php?show_all=1) download page
- Chose the download link appropriate to your system
- To use it, double-click on the downloaded file
- Follow any installation instructions, if any
- At the workshop we will give you additional information (e.g. host, username, port) to connect

### Chrome
- Go to the [Chrome download page](https://www.google.com/chrome/browser/desktop/index.html)
- Click and the button and install Chrome
<!-- 
### Dendroscope
- Download Dendroscope 3 [here](http://ab.inf.uni-tuebingen.de/data/software/dendroscope3/download/welcome.html).
- Chose the download link appropriate to your system
- To use it, double-click on the downloaded file
- Follow any installation instructions, if any
-->
### IGV
We will use The Broad Institute's IGV (Integrated Genome Viewer) for looking at SAM and BAM files, SNPs, and variant calls.
 - To download IGV, please visit this page and log in
 - Click on the large button for the Download Binary Distribution
 - Find the zip archive on your computer and expand it
 - To run IGV, double-click on the IGV.bat file. 




{% include links.md %}
