---
title: "Sequence assembly"
teaching: 45
exercises: 30
questions:
- "How can the information in the sequencing reads be reduced?"
- "What are the different methods for assembly?"
objectives:
- "Understand differences between assembly methods"
- "Assemble the short reads"
keypoints:
- "Assembly is a process which aligns and merges fragments from a longer DNA sequence in order to reconstruct the original sequence."
- "k-mers are short fragments of DNA of length k"
---

Sequence assembly means the alignment and merging of reads in order to reconstruct the original sequence. We will talk about sequence assembly in a separate [lecture](https://www.dropbox.com/s/89x5wrdzigegr6q/Assembly%20introduction%202018.pptx?dl=0). The assembly of a genome from short sequencing reads will take a while - up to several hours per genome. We will therefore run this process overnight in the background. 

To this end we will again start a screen session. It will allow us to run our entire shell and keep that process running even if we disconnect.

## Starting and attaching to screen sessions

### Starting a new session

You can start a session and give it a descriptive name:

~~~
$ screen -S assembly
~~~
{: .bash}

This creates a session with the name ‘assembly’

As you work, this session will stay active until you close it. Even if you log out or work on something else, the jobs you start in this session will run until completion.


## Sequence assembly

The assembler we will run is [SPAdes](http://cab.spbu.ru/software/spades/). SPAdes generates a final assembly from multiple kmers. A list of kmers is automatically selected by SPAdes using the maximum read length of the input data, and each individual kmer contributes to the final assembly. If desired, a list of kmers can be specified with the -k flag which will override automatic kmer selection.


## Assembly (over night)

Because assembly of each genome might take a couple of hours, we will run all assemblies in a loop overnight. It is important to run them within the screen session or else the process will be terminated if we disconnect from the machine. 

Preparation
~~~
$ mkdir ~/dc_workshop/results/assembly
$ cd ~/dc_workshop/data/trimmed_fastq
~~~
{: .bash}

To run SPAdes we will use the spades.py command with the --careful option to minimize the number of mismatches in the contigs, -o for the output folder, -1 for the path to the forward reads, -2 for the path to the reverse reads, and -s for the path to the singles reads. We can start the loop with the assemblies

~~~
$ for sample in ERR026473 ERR026474 ERR026478 ERR026481 ERR026482 ERR029206 ERR029207
>  do
>  spades.py  --careful -1 "$sample"_1.fastq_trim.fastq -2 "$sample"_2.fastq_trim.fastq -o ../results/assembly/"$sample"
>  done
~~~
{: .bash}

**Detach session (process keeps running in background)**

You can detach from a session by pressing `control + a` followed by `d` (for detach) on your keyboard. 

You can now safely work on something else. The assembly will keep on running.


> ## Challenge: How many contigs were generated by SPAdes??
>
> Find out how many contigs there are in the *M. tuberculosis* isolates. Enter your solution in the
> [table](https://docs.google.com/spreadsheets/d/1xjiliy_USyMwiyzEgWhpn8_109F7Z3jPM_f7Jp-lOb8/edit?usp=sharing)
>
> Hint:
> ~~~
> $ grep -c
> ~~~
> prints a count of matching lines for each input file.
> 
> > ## Solution
> >
> > 
> > ~~~
> > $ grep -c '>'  ERR02*/contigs.fasta
> > 
> > ERR026473/contigs.fasta:382
> > ERR026474/contigs.fasta:346
> > ERR026478/contigs.fasta:315
> > ERR026481/contigs.fasta:363
> > ERR026482/contigs.fasta:371
> > ERR029206/contigs.fasta:303
> > ERR029207/contigs.fasta:352
> > ~~~
> > {: .output}
> {: .solution}
{: .challenge}


At the moment, all samples are called contigs.fasta. This is not ideal. In the next episode we will rename the assembled contigs before processing them further.


{% include links.md %}
