---
title: "Transmission?"
teaching: 5
exercises: 20
questions:
- "In which cases is transmission likely?"
objectives:
- "Compare and interpret the different data sources."
keypoints:
- "SNP phylogeny, pangenome analysis and metadata can convey different messages"
- "Human interpretation is often needed to weigh the different information sources."
- "The low mutation rate of *M. tuberculosis* does not allow to make confident inferences of transmission but does allow to exclude transmission" 
---

The low mutation rate of *M. tuberculosis* means that even at the highest resolution provided by whole genome sequencing it is still difficult to confidently affirm the inferences of transmission made by traditional epidemiological techniques. This means it is very difficult to determine transmission inclusively. However, whole genome sequencing does in some cases allow us to exclude direct transmission, by using the phylogenetic context provided by other strains. Not only does whole genome sequencing provide the inter-cluster differentiation provided by current typing methods, but it also achieves intra-cluster resolution which can be used to inform epidemiological investigation.


> ## Discussion
>
>
> Compare the pangenome clustering with the clustering based on SNPs or gene presence/absence and with the data from 
> epidemiological contact tracing:
> 
> Cluster A
> - Patient A1 - sample ERR029207
> - Patient A2 - sample ERR029206
> - Patient A3 - sample ERR026478
> 
> Cluster B
> - Patient B1 - sample ERR026474 
> - Patient B2 - sample ERR026473
>
> Cluster C
> - Patient C1 - sample ERR026481
> - Patient C2 - sample ERR026482
> 
> Which transmission events are affirmed by the genomic data? Which ones partially or not? Why?
>
{: .discussion}
